stl/inc/algorithm

```diff
@@ -8004,6 +8541,323 @@ void stable_sort(_ExPo&& _Exec, _BidIt _First, _BidIt _Last) noexcept /* termina
+                                             .out;
+                        return _Backout._Release();
+                    }
+                } else {
+                    _Backout._Emplace_back(_RANGES iter_move(_First));
+                    ++_First;
+
+                    if (_First == _Mid) {
+                        _Backout._Last = _RANGES _Uninitialized_move_unchecked(
+                            _STD move(_Next), _STD move(_Last), _Backout._Last, _Dest_last)
+                                             .out;
+                        return _Backout._Release();
+                    }
+                }
+            }
+        }
+
+        template <class _It, class _Pr, class _Pj>
+        static void _Chunked_merge_to_buffer(_It _First, const _It _Last, iter_value_t<_It>* _Dest,
+            const iter_difference_t<_It> _Chunk_size, iter_difference_t<_It> _Count, _Pr _Pred, _Pj _Proj) {
+            // move merging adjacent chunks of distance _Chunk_size to temporary buffer
+            _STL_INTERNAL_STATIC_ASSERT(random_access_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(sortable<_It, _Pr, _Pj>);
+            _STL_INTERNAL_CHECK(_RANGES distance(_First, _Last) == _Count);
+            _STL_INTERNAL_CHECK(_Chunk_size > 0);
+
+            while (_Chunk_size < _Count) {
+                _Count -= _Chunk_size;
+                const auto _Right_chunk_size = (_STD min)(_Chunk_size, _Count);
+                _Count -= _Right_chunk_size;
+
+                const auto _Mid1  = _First + _Chunk_size;
+                const auto _Last1 = _Mid1 + _Right_chunk_size;
+                _Dest             = _Merge_move_to_buffer(_First, _Mid1, _Last1, _Dest, _Pred, _Proj);
+                _First            = _Last1;
+            }
+
+            // copy partial last chunk
+            _RANGES _Move_unchecked(_STD move(_First), _STD move(_Last), _STD move(_Dest));
+        }
+
+        template <class _It, class _Pr, class _Pj>
+        _NODISCARD static iter_value_t<_It>* _Merge_move_to_buffer(
+            _It _First, _It _Mid, _It _Last, iter_value_t<_It>* const _Dest, _Pr _Pred, _Pj _Proj) {
+            // move merging ranges to temporary storage
+            _STL_INTERNAL_STATIC_ASSERT(random_access_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(sortable<_It, _Pr, _Pj>);
+            _STL_INTERNAL_CHECK(_First != _Mid);
+            _STL_INTERNAL_CHECK(_Mid != _Last);
+
+            _Uninitialized_backout<iter_value_t<_It>*> _Backout{_Dest};
+            _It _Next = _Mid;
+            for (;;) {
+                if (_STD invoke(_Pred, _STD invoke(_Proj, *_Next), _STD invoke(_Proj, *_First))) {
+                    _RANGES iter_swap(_Backout._Last, _Next);
+                    ++_Backout._Last;
+                    ++_Next;
+
+                    if (_Next == _Last) {
+                        _Backout._Last =
+                            _RANGES _Move_unchecked(_STD move(_First), _STD move(_Mid), _Backout._Last).out;

```

Comment:
Ditto I'm confused about whether or not we expect `_Dest` to point to live objects.

Reply:
As said above the should be in a moved-from state